### PR TITLE
Add table of contents and documentation index to README file

### PR DESCRIPTION
**Pull request by MostafaMamouni**
This pull request is about adding more accessibility to the README and the user docs by having a `table of contents` for the content of README and also an `index` for the user docs.

### CODE REVIEW

**Comment by EmilyBourne** [comment]
I think this looks like a really good way to make the first look at Pyccel more user-friendly. Great job !

File: README.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+-   [Header-files](./tutorial/header-files.md)
+
+-   [Headers](./tutorial/headers.md)
+
+-   [Ndarrays](./tutorial/ndarrays.md)
+
+-   Supported libraries/APIs
+    -   [OpenMP](./tutorial/openmp.md)
+    -   [Numpy](./tutorial/numpy-functions.md)
```
[the remark this answers is not in the document]
Comment (yguclu) [reply README.md]: cc: @MostafaMamouni

**Review by EmilyBourne — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+Finally, Pyccel supports distributed-memory parallel programming through the Message Passing Interface (MPI) standard; hence it requires an MPI library like
+
+-   [Open-MPI](https://www.open-mpi.org/)
+-   [MPICH](https://www.mpich.org/)
+-   [Intel® MPI Library](https://software.intel.com/en-us/mpi-library)
+
+We recommend using GFortran/GCC and Open-MPI.
+
+Pyccel also depends on several Python3 packages, which are automatically downloaded by pip, the Python Package Installer, during the installation process. In addition to these, unit tests require the _scipy_, _mpi4py_, _pytest_ and _coverage_ packages, while building the documentation requires [Sphinx](http://www.sphinx-doc.org/).
```
Comment (EmilyBourne) [inline README.md:103]: I've just noticed that we don't mention OpenMP in the requirements. Could you add it please

File: README.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+### Linux Debian-Ubuntu-Mint
+
+To install all requirements on a Linux Ubuntu machine, just use APT, the Advanced Package Tool:
+
+```sh
+sudo apt update
+sudo apt install gcc
+sudo apt install gfortran
+sudo apt install libblas-dev liblapack-dev
+sudo apt install libopenmpi-dev openmpi-bin
```
Comment (EmilyBourne) [inline README.md:114]: sudo apt-get install libomp-dev libomp5

File: README.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+Install all requirements using the DNF software package manager:
+
+```sh
+su
+dnf check-update
+dnf install gcc
+dnf install gfortran
+dnf install blas-devel lapack-devel
+dnf install openmpi-devel
```
Comment (EmilyBourne) [inline README.md:127]: I don't know who runs Fedora and can provide the install command :/

File: README.md [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+### Mac OS X
+
+On an Apple Macintosh machine we recommend using [Homebrew](https://brew.sh/):
+
+```sh
+brew update
+brew install gcc
+brew install openblas
+brew install lapack
+brew install open-mpi
```
Comment (EmilyBourne) [inline README.md:142]: brew install libomp

**Review by yguclu — APPROVED** [review]

**Pull request merged**